src/main/kotlin/com/wutsi/application/shell/endpoint/settings/profile/command/UpdateProfileCommand.kt

```diff
@@ -0,0 +1,35 @@
+package com.wutsi.application.shell.endpoint.settings.profile.command
+
+import com.wutsi.application.shell.endpoint.AbstractCommand
+import com.wutsi.application.shell.endpoint.settings.profile.dto.UpdateProfileRequest
+import com.wutsi.application.shell.service.AccountService
+import com.wutsi.flutter.sdui.Action
+import com.wutsi.flutter.sdui.enums.ActionType
+import org.springframework.http.HttpHeaders
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RestController
+
+@RestController
+@RequestMapping("/commands/update-profile")
+class UpdateProfileCommand(
+    private val accountService: AccountService
+) : AbstractCommand() {
+    @PostMapping
+    fun index(@RequestBody request: UpdateProfileRequest): ResponseEntity<Action> {
+        accountService.updateProfile(request)
+        val headers = HttpHeaders()
+        headers["x-language"] = request.language
```

Comment:
Returning the language in the header so that the flutter app will store it into the preference on the device.
Then the language is added in all request to BFF via the header `Accept-Language`, this information will be used by BFF to return text in the proper language